OpenGL: Point rendering not working

### Godot version

4.0.rc3

### System information

Fedora 37, GeForce RTX 4090 (NVIDIA 525.85.05)

### Issue description

Enabling **Transform > Use Point Size** and changing **Transform > Point Size** in a BaseMaterial3D has no effect when using the Compatibility rendering method, but it works when using Forward+ or Forward Mobile (Vulkan).

themoleman on the Godot Contributors Chat has hinted the following:

> Just a FYI there is a bug in the compatability renderer that is preventing the gl_PointSize from being set
> I ran apitrace on godot and got the following error:
> warning C7050 `gl_pointSize` might be used before being initialized

### Forward+

*Point sizes of 1, 6 and 20 are used respectively. The red points on the right use a transparent textured material.*

![Screenshot_20230223_174332](https://user-images.githubusercontent.com/180032/220973448-44a1fdb0-ab55-40c3-b665-7dba8847c201.png)

### Mobile

*Shadow casting appears different due to https://github.com/godotengine/godot/issues/73827.*

![Screenshot_20230223_174348](https://user-images.githubusercontent.com/180032/220973466-bf4d7d2a-b5c6-4a39-95ce-19421b7e209a.png)

### Compatibility

*Red sphere is invisible because of the transparent texture that spans the entire sphere, as opposed to individual points.*

![Screenshot_20230223_174401](https://user-images.githubusercontent.com/180032/220973489-19d4d03a-87e9-4b85-b078-53db15c3892e.png)

### Steps to reproduce

- Create a MeshInstance3D with a SphereMesh.
- Add a BaseMaterial3D with **Transform > Use Point Size** enabled, then change **Transform > Point Size**.
- Notice this has no effect if the editor is currently running with the Compatibility rendering method.

### Minimal reproduction project

[test_point_rendering.zip](https://github.com/godotengine/godot/files/10816063/test_point_rendering.zip)

--- a/drivers/gles3/rasterizer_scene_gles3.cpp
+++ b/drivers/gles3/rasterizer_scene_gles3.cpp
@@ -2206,6 +2206,9 @@ void RasterizerSceneGLES3::_render_list_template(RenderListParameters *p_params,
 		}
 
 		RS::PrimitiveType primitive = surf->primitive;
+		if (shader->uses_point_size) {
+			primitive = RS::PRIMITIVE_POINTS;
+		}
 		static const GLenum prim[5] = { GL_POINTS, GL_LINES, GL_LINE_STRIP, GL_TRIANGLES, GL_TRIANGLE_STRIP };
 		GLenum primitive_gl = prim[int(primitive)];
 

--- a/drivers/gles3/storage/material_storage.cpp
+++ b/drivers/gles3/storage/material_storage.cpp
@@ -1494,7 +1494,7 @@ MaterialStorage::MaterialStorage() {
 		actions.renames["LIGHT_VERTEX"] = "light_vertex";
 		actions.renames["SHADOW_VERTEX"] = "shadow_vertex";
 		actions.renames["UV"] = "uv";
-		actions.renames["POINT_SIZE"] = "gl_PointSize";
+		actions.renames["POINT_SIZE"] = "point_size";
 
 		actions.renames["MODEL_MATRIX"] = "model_matrix";
 		actions.renames["CANVAS_MATRIX"] = "canvas_transform";
@@ -1575,7 +1575,7 @@ MaterialStorage::MaterialStorage() {
 		actions.renames["UV"] = "uv_interp";
 		actions.renames["UV2"] = "uv2_interp";
 		actions.renames["COLOR"] = "color_interp";
-		actions.renames["POINT_SIZE"] = "gl_PointSize";
+		actions.renames["POINT_SIZE"] = "point_size";
 		actions.renames["INSTANCE_ID"] = "gl_InstanceID";
 		actions.renames["VERTEX_ID"] = "gl_VertexID";
 

